test/useAsync.test.ts

```diff
@@ -0,0 +1,85 @@
+import { useAsync, UseAsyncReturn } from '../src';
+import { renderHook } from '@testing-library/react-hooks';
+
+interface StarwarsHero {
+  name: string;
+}
+
+type StarwarsHeroArgs = {
+  asyncFunction: () => Promise<StarwarsHero[]>;
+};
+
+export const generateMockResponseData = (amount: number = 5): StarwarsHero[] =>
+  [...Array(amount).keys()].map(n => ({
+    id: n + 1,
+    name: `Starwars Hero ${n + 1}`,
+  }));
+
+describe('useAync', () => {
+  const fakeResults = generateMockResponseData();
+
+  it('should have a useAsync hook', () => {
+    expect(useAsync).toBeDefined();
+  });
+
+  it('should resolve a successful request', async () => {
+    const onSuccess = jest.fn();
+    const onError = jest.fn();
+
+    const { result, waitForNextUpdate } = renderHook<
+      StarwarsHeroArgs,
+      UseAsyncReturn<StarwarsHero[]>
+    >(() =>
+      useAsync<StarwarsHero[], any>(
+        async () => {
+          return Promise.resolve(fakeResults);
+        },
+        [],
+        {
+          onSuccess: () => onSuccess(),
+          onError: () => onError(),
+        }
+      )
+    );
+
+    expect(result.current.loading).toBe(true);
+
+    await waitForNextUpdate();
+
+    expect(result.current.result).toEqual(fakeResults);
+    expect(result.current.loading).toBe(false);
+    expect(result.current.error).toBeUndefined();
+    expect(onSuccess).toHaveBeenCalled();
+    expect(onError).not.toHaveBeenCalled();
+  });
+
+  it('should set error detail for unsuccessful request', async () => {
+    const onSuccess = jest.fn();
+    const onError = jest.fn();
+
+    const { result, waitForNextUpdate } = renderHook<
+      StarwarsHeroArgs,
+      UseAsyncReturn<StarwarsHero[]>
+    >(() =>
+      useAsync<StarwarsHero[], any>(
+        async () => {
+          return Promise.reject(new Error('something went wrong'));
```

Comment:
I think you can just throw new Error("") here, that would be the same

Reply:
I can't get it to work with just returning the error.  Please feel free to change this